gnocchi/incoming/redis.py

```diff
@@ -108,3 +108,15 @@ def process_measure_for_metric(self, metric):
 
         # ltrim is inclusive, bump 1 to remove up to and including nth item
         self._client.ltrim(key, item_len + 1, -1)
+
+    def iter_on_sacks_to_process(self):
+        self._client.config_set("notify-keyspace-events", "El")
+        p = self._client.pubsub()
+        db = self._client.connection_pool.connection_kwargs['db']
+        channel = b"__keyevent@" + str(db).encode() + b"__:rpush"
```

Comment:
the above should be part of init? we shouldn't need to do it constantly or does the p.listen() never exit?

Reply:
If any exception occurs, you need to reconnect and resubscribe. The reconnection is handled by the library but then it needs to recreate a publisher.

If everything goes fine, this is executed only once. Otherwise, it re-sub.